apps/openmw/mwrender/renderingmanager.cpp

```diff
@@ -97,6 +97,7 @@ namespace MWRender
             stateset->setAttribute(lightModel, osg::StateAttribute::ON);
             osg::Fog* fog = new osg::Fog;
             fog->setMode(osg::Fog::LINEAR);
+            fog->setUseRadialFog(true);
```

Comment:
You should probably have this controlled by the setting we're using elsewhere. The extension is proprietary to Nvidia, so no one else can implement it even if they wanted to.

This does mean, though, that if `osg::isGLExtensionSupported(state.getContextID(),"GL_NV_fog_distance");` returns true, you don't actually have to force-enable shaders when the setting is on.

Reply:
> The extension is proprietary to Nvidia, so no one else can implement it even if they wanted to.

That isn't true, Mesa implements [[1]](https://www.mail-archive.com/mesa-dev@lists.freedesktop.org/msg11988.html)[[2]](https://cgit.freedesktop.org/mesa/mesa/log/?qt=grep&q=NV_fog_distance) this setting for all Gallium drivers. See also [Apple's data](http://arek.bdmonkeys.net/GLInfo/GLInfo_1068.html) for their proprietary Radeon driver.

Which setting do you have in mind? Should it be "accurate fog" as in this PR?

Reply:
> > The extension is proprietary to Nvidia, so no one else can implement it even if they wanted to.
> 
> That isn't true, Mesa implements [[1]](https://www.mail-archive.com/mesa-dev@lists.freedesktop.org/msg11988.html)[[2]](https://cgit.freedesktop.org/mesa/mesa/log/?qt=grep&q=NV_fog_distance) this setting for all Gallium drivers. See also [Apple's data](http://arek.bdmonkeys.net/GLInfo/GLInfo_1068.html) for their proprietary Radeon driver.

Usually when an extension says it's proprietary to one vendor in its text, other vendors have to pay a fee to implement it. I imagine Apple paid to be allowed to do it, and have no idea how Mesa is allowed (maybe Nvidia granted them blanket permissions to do whatever they want, or maybe Nvidia just doesn't sue them). I hadn't seen some of that when I wrote my post, though, so assumed no one had bothered to get permission. It's definitely not supported by AMD hardware on Windows, though. I tested that myself.

> Which setting do you have in mind? Should it be "accurate fog" as in this PR?

Yeah, it should come under the same setting

Reply:
> Regardless of the vendor prefix, vendor-specific OpenGL extensions may be implemented an/or licensed by multiple hardware vendors to ensure compatibility. This is very rare in practice.

For example: GL_NV_texture_barrier (implemented widely on AMD hardware)

https://mesamatrix.net/ <-- complete coverage of GL extensions for mesa supported hardware

Reply:
Did more research, here is the result:

> Name NV_fog_distance
> Name Strings GL_NV_fog_distance
> Notice Copyright NVIDIA Corporation, 1999, 2000, 2001.
> IP Status  NVIDIA Proprietary.
> Status  Shipping (version 1.0)
> Version NVIDIA Date: January 18, 2001
>  $Id: //sw/main/docs/OpenGL/specs/GL_NV_fog_distance.txt#14 $ 

http://developer.download.nvidia.com/opengl/specs/nvOpenGLspecs.pdf

However, it was implimented for Mesa in 2011 here:
https://lists.freedesktop.org/archives/mesa-commit/2011-November/033899.html
discussion about it:
https://www.mail-archive.com/mesa-dev@lists.freedesktop.org/msg11929.html

It is like I said above, their way of implimenting it is NVIDIA Proprietary, but if someone implements it themselves without looking at how NVIDIA does it, then it's OK. Just like S2TC.